plugins/attachment-resources/src/components/LinkPreviewPresenter.svelte

```diff
@@ -0,0 +1,74 @@
+<!-- //
+// Copyright © 2025 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -->
+<script lang="ts">
+  import { type LinkPreviewDetails } from '@hcengineering/presentation'
+  import { fetchJson } from '@hcengineering/presentation'
+  import { type Attachment } from '@hcengineering/attachment'
+  import { type WithLookup } from '@hcengineering/core'
+  import { Spinner } from '@hcengineering/ui'
+  export let attachment: WithLookup<Attachment>
+
+  async function fetchViewModel (): Promise<LinkPreviewDetails> {
```

Comment:
rename to `fetchLinkPreview`?
You should pass `attachment` as an argument to ensure Svelte rerenders the component if `attachment` changes.